### PR TITLE
TimeSeries

### PR DESCRIPTION
**Pull request by scopatz**
This implements a time series convention helpers.

### CODE REVIEW

File: tests/toolkit/timeseries_tests.cc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  RecordTimeSeries<POWER>(a, 42.0);
+  delete a;
+}
+
+TEST(TimeSeriesTests, RawPower) {
+  TestContext tc;
+  Agent* a = new TestAgent(tc.get());
+  RecordTimeSeries<double>("Power", a, 42.0);
+  delete a;
+}
```
Comment (mbmcgarry) [inline tests/toolkit/timeseries_tests.cc]: Are you looking at the output database separately to see that the information was actually recorded?

File: src/toolkit/timeseries.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+#include <string>
+
+#include "agent.h"
+#include "context.h"
+
+namespace cyclus {
+namespace toolkit {
+
+/// Time series types to be used in the RecordTimeSeries() fucntions.
```
Comment (mbmcgarry) [inline src/toolkit/timeseries.h]: should be functions

File: tests/toolkit/timeseries_tests.cc (hunk 1 above)
Comment (scopatz) [inline tests/toolkit/timeseries_tests.cc]: No, this doesn't write out to a real database. It just implements the correct interface for testing.  This strategy is used in other parts of the test suite. We test database writing in other places as well.

File: src/toolkit/timeseries.h (hunk 2 above)
Comment (scopatz) [inline src/toolkit/timeseries.h]: Thanks!

File: tests/toolkit/timeseries_tests.cc (hunk 1 above)
Comment (mbmcgarry) [inline tests/toolkit/timeseries_tests.cc]: Looks reasonable from my limited perspective.

**Comment by scopatz** [comment]
I think this is good to go, but did anyone else want to take a look?

**Comment by rwcarlsen** [comment]
I'd like to take a closer look tomorrow morning if you don't mind waiting
 until then.

**Comment by scopatz** [comment]
I think that we'll be hitting the need for this tomorrow afternoon.  So before then would be great

File: src/toolkit/timeseries.cc [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#include "timeseries.h"
+
+namespace cyclus {
+namespace toolkit {
+
+template <>
+void RecordTimeSeries<POWER>(cyclus::Agent* agent, double value) {
+  RecordTimeSeries<double>("Power", agent, value);
```
Comment (rwcarlsen) [inline src/toolkit/timeseries.cc]: Not a stopper, but FYI - you could just `RecordTimeSeries("Power", agent, value);` and the compiler will infer the templated type for the function.

File: src/toolkit/timeseries.h [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+#include "agent.h"
+#include "context.h"
+
+namespace cyclus {
+namespace toolkit {
+
+/// Time series types to be used in the RecordTimeSeries() functions.
+/// These types have the following unit which *must* be adhered to strictly:
+/// - POWER [MWe]
+enum TimeSeriesTypes {
```
Comment (rwcarlsen) [inline src/toolkit/timeseries.h]: This should be singular - it makes the it feel more natural when defining/using variables of this type - e.g.

```c++
TimeSeriesTypes mytype; // this is awkward

TimeSeriesType mytype; // more natural
```

File: tests/toolkit/timeseries_tests.cc (hunk 1 above)
Comment (rwcarlsen) [inline tests/toolkit/timeseries_tests.cc]: While we shouldn't be (re)testing the database writing/querying here, I will note that these tests are not particularly useful - they test nothing about the functions except that they can be called without throwing exceptions or segfaulting - which IMO, is hardly better than having no tests at all w.r.t. how much tests can actually do.

File: src/toolkit/timeseries.cc (hunk 3 above)
Comment (scopatz) [inline src/toolkit/timeseries.cc]: "explicit is better than implicit," or so they tell me :)

File: src/toolkit/timeseries.h (hunk 4 above)
Comment (scopatz) [inline src/toolkit/timeseries.h]: Yup you are right.

File: tests/toolkit/timeseries_tests.cc (hunk 1 above)
Comment (scopatz) [inline tests/toolkit/timeseries_tests.cc]: In my opinion, the existence of a valid test is always better than no test, irrespective of the depth of functionality it tests. Unit tests in particular, should aim to test as simple of a behavior as possible.  I think that it is also worth noting that these underlying functions don't actually do that much and aren't that big.  They exist to establish a convention, not implement a lot of behavior.  I think that this is actually a pretty decent example of a small function with small tests implemented via mocked behavior.

**Comment by scopatz** [comment]
OK! I think this is ready. Thanks for the review.

**Pull request merged**